framework/tlvf/src/include/tlvf/WSC/m1.h

```diff
@@ -16,11 +16,35 @@ namespace WSC {
 class m1 : public AttrList {
 
 public:
+    struct config {
+        eWscMessageType msg_type;
+        uint8_t uuid_e[WSC_UUID_LENGTH];
+        uint8_t uuid_r[WSC_UUID_LENGTH];
+        sMacAddr mac;
+        uint8_t enrollee_nonce[WSC_NONCE_LENGTH];
+        uint8_t registrar_nonce[WSC_NONCE_LENGTH];
+        uint8_t pub_key[WSC_PUBLIC_KEY_LENGTH];
+        uint16_t encr_type_flags;
+        uint16_t auth_type_flags;
+        std::string manufacturer;
+        std::string model_name;
+        std::string model_number;
+        std::string serial_number;
+        uint16_t primary_dev_type_id;
+        std::string device_name;
+        eWscRfBands bands;
```

Comment:
I would have kept the common part (i.e. this bit) in the AttrList class, as well as the getter wrappers for them, and I'd have kept a protected AttrList::create() function that takes the common config. I would have passed the additional configuration as separate parameters to the (public) m1 and m2 create() functions (so the two would have a different signature).

However, after seeing the configData, I kind of like how you did it: every type of attribute list has a number of expected attributes, and you can provide all of them to the factory method (create()) in one shot, and the structure is always called `config`.